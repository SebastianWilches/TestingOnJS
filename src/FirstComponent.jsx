import PropTypes from 'prop-types'

export const FirstComponent = ({ titulo, parrafo, suma }) => {
    return (
        <>
            <h1>{titulo}</h1>
            <p>{parrafo}</p>
            <p>{suma + 30}</p>
        </>
    )
}

FirstComponent.propTypes = {
    titulo: PropTypes.string.isRequired, //El titulo tiene que ser un String requerido
    parrafo: PropTypes.string, //El parrafo tiene que ser String
    suma: PropTypes.number
}

FirstComponent.defaultProps = {
    titulo: "No hay titulo",
    parrafo: "No hay informaci√≥n de parrafo",
    suma: 1000
}