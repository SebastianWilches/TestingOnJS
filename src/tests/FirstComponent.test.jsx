import { render } from '@testing-library/react'
import { FirstComponent } from "../FirstComponent"

describe('Pruebas en el componente <FirstComponent/>', () => {

    test('Debe hacer match con el snapshot', () => {
        //Arrange
        const title = 'Titulo de la app';
        const parrafo = 'LoremIpsum';
        const suma = 12;
        //Act
        const { container } = render(<FirstComponent titulo={title} parrafo={parrafo} suma={suma} />);

        //Affirm
        expect(container).toMatchSnapshot();

    })

    test('El H1 debería mostrar información del titulo', () => { 
        //A
        const title = 'Titulo de la app';

        //A
        const { container, getByText } = render(<FirstComponent titulo={title}/>);

        //A
        expect(getByText(title)).toBeTruthy(); //Evaluamos que el texto este en el DOM

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title); //Miramos que un elemento h1 tenga el titulo

     })

})