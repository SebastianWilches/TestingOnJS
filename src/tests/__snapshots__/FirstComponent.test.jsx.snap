// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <FirstComponent/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Titulo de la app
  </h1>
  <p>
    LoremIpsum
  </p>
  <p>
    42
  </p>
</div>
`;
